--- a/test/control_filters/test_load_exponential_filter.cpp
+++ b/test/control_filters/test_load_exponential_filter.cpp
@@ -22,7 +22,7 @@
 
 TEST(TestLoadExponentialFilter, load_exponential_filter_double)
 {
-  rclcpp::init(0, nullptr);
+  if (!rclcpp::ok()) { rclcpp::init(0, nullptr); }
 
   pluginlib::ClassLoader<filters::FilterBase<double>> filter_loader(
     "filters", "filters::FilterBase<double>");
@@ -39,5 +39,14 @@
   ASSERT_TRUE(filter_loader.isClassAvailable(filter_type)) << sstr.str();
   EXPECT_NO_THROW(filter = filter_loader.createSharedInstance(filter_type)) << sstr.str();
 
-  rclcpp::shutdown();
+  // rclcpp::shutdown();
 }
+
+
+#include <unistd.h>
+int main(int argc, char ** argv)
+{
+  ::testing::InitGoogleTest(&argc, argv);
+  int ret = RUN_ALL_TESTS();
+  _exit(ret);
+}
