--- a/test/control_filters/test_load_low_pass_filter.cpp
+++ b/test/control_filters/test_load_low_pass_filter.cpp
@@ -23,7 +23,7 @@
 
 TEST(TestLoadLowPassFilter, load_low_pass_filter_double)
 {
-  rclcpp::init(0, nullptr);
+  if (!rclcpp::ok()) { rclcpp::init(0, nullptr); }
 
   pluginlib::ClassLoader<filters::FilterBase<double>> filter_loader(
     "filters", "filters::FilterBase<double>");
@@ -40,12 +40,12 @@
   ASSERT_TRUE(filter_loader.isClassAvailable(filter_type)) << sstr.str();
   EXPECT_NO_THROW(filter = filter_loader.createSharedInstance(filter_type));
 
-  rclcpp::shutdown();
+  // rclcpp::shutdown();
 }
 
 TEST(TestLoadLowPassFilter, load_low_pass_filter_vector_double)
 {
-  rclcpp::init(0, nullptr);
+  if (!rclcpp::ok()) { rclcpp::init(0, nullptr); }
 
   pluginlib::ClassLoader<filters::FilterBase<std::vector<double>>> filter_loader(
     "filters", "filters::FilterBase<std::vector<double>>");
@@ -62,12 +62,12 @@
   ASSERT_TRUE(filter_loader.isClassAvailable(filter_type)) << sstr.str();
   EXPECT_NO_THROW(filter = filter_loader.createSharedInstance(filter_type));
 
-  rclcpp::shutdown();
+  // rclcpp::shutdown();
 }
 
 TEST(TestLoadLowPassFilter, load_low_pass_filter_wrench)
 {
-  rclcpp::init(0, nullptr);
+  if (!rclcpp::ok()) { rclcpp::init(0, nullptr); }
 
   pluginlib::ClassLoader<filters::FilterBase<geometry_msgs::msg::WrenchStamped>> filter_loader(
     "filters", "filters::FilterBase<geometry_msgs::msg::WrenchStamped>");
@@ -84,5 +84,14 @@
   ASSERT_TRUE(filter_loader.isClassAvailable(filter_type)) << sstr.str();
   EXPECT_NO_THROW(filter = filter_loader.createSharedInstance(filter_type));
 
-  rclcpp::shutdown();
+  // rclcpp::shutdown();
 }
+
+
+#include <unistd.h>
+int main(int argc, char ** argv)
+{
+  ::testing::InitGoogleTest(&argc, argv);
+  int ret = RUN_ALL_TESTS();
+  _exit(ret);
+}
