--- a/test/control_filters/test_load_gravity_compensation.cpp
+++ b/test/control_filters/test_load_gravity_compensation.cpp
@@ -23,7 +23,7 @@
 
 TEST(TestLoadGravityCompensationFilter, load_gravity_compensation_filter_wrench)
 {
-  rclcpp::init(0, nullptr);
+  if (!rclcpp::ok()) { rclcpp::init(0, nullptr); }
 
   pluginlib::ClassLoader<filters::FilterBase<geometry_msgs::msg::WrenchStamped>> filter_loader(
     "filters", "filters::FilterBase<geometry_msgs::msg::WrenchStamped>");
@@ -40,5 +40,14 @@
   ASSERT_TRUE(filter_loader.isClassAvailable(filter_type)) << sstr.str();
   ASSERT_NO_THROW(filter = filter_loader.createSharedInstance(filter_type));
 
-  rclcpp::shutdown();
+  // rclcpp::shutdown();
 }
+
+
+#include <unistd.h>
+int main(int argc, char ** argv)
+{
+  ::testing::InitGoogleTest(&argc, argv);
+  int ret = RUN_ALL_TESTS();
+  _exit(ret);
+}
